sirepo/srschema.py

```diff
@@ -200,6 +202,16 @@ def _validate_enum(val, sch_field_info, sch_enums):
     if str(val) not in map(lambda enum: str(enum[0]), sch_enums[type]):
         raise AssertionError(util.err(sch_enums, 'enum {} value {} not in schema', type, val))
 
+def _validate_model_name(model_name):
+    """Ensure model name contain no special characters
+
+    Args:
+        model_name (str): name to validate
+    """
+
+    m = re.findall(r'[^\w]', model_name)
```

Comment:
`\W` can be used in place of `[^\w]`. 

I think there is little value in printed the invalid characters if you are printing the model name. I think instead the error string should say "model name must contain word characters only".  Therefore, you don't the findall, just a search.

Question: do we want to limit model names to `^[a-z]\W*$`. If so, you could say "must be a Python identifier" in error.

Reply:
Ah I didn't know about the uppercase inverse in regex. Thanks.

Yes I think a python identifier is better. That way PKDict lookup using dot notation will always work and we still solve the other problems.

I'm not sure about `^[a-z]\W*$`. For example we have [_APERTURE_ELEMENT](https://github.com/radiasoft/sirepo/blob/71f3056a6acaf6519761aef9fdae7de7bc238f7b/sirepo/package_data/static/json/madx-schema.json#L515) which doesn't pass that regex. How about `^[^\d\W]\w*$`?

Reply:
I think positive is better `^[a-z_]\w*$` with ignore case on the regex.

Reply:
Done https://github.com/radiasoft/sirepo/commit/a846d89d5ca063e9944da7578e4adcba94194a7a